### PR TITLE
New backend implementation and massaging data seeding

### PR DESCRIPTION
**Pull request by IamTechknow**

### CODE REVIEW

**Comment by IsaacLeeWebDev** [comment]
Sorry to keep you waiting! I'll go ahead and review this one.

File: README.md [hunk 1]
```diff
@@ -7,23 +7,27 @@
...
 ## CRUD Documentation
 
 ### Data Shape
-  - Campaign: Campaign name, description, author, currency, pledge amount, pledge goal, backers, end date, type, location
+  - Campaign: Campaign name, description, author, username, country, pledge amount, pledge goal, backers, end date, type
   - Pledge: username, pledge amount, date
 
 ### Page Endpoint
-/:author/:campaignId/
+/:user/:campaignId/
```
Comment (IzzyLeeDev) [inline README.md:21]: Is this a functional change, or a conventional change? There's no problem either way, I just want to make sure I'm on board with my conventions.

File: README.md [hunk 2]
```diff
@@ -7,23 +7,27 @@
...
 
 ### GET - /api/:campaign/stats
   - Get stats from a single campaign, for an item detail view.
   - Query format: Campaign ID (number)
   - Response data shape: Refer to Campaign data shape above
   - Error code: 400 for invalid author or campaign
 
-### GET - /api/:author/campaigns
-  - Get all campaigns for given author
+### GET - /api/:user/campaigns
```
Comment (IzzyLeeDev) [inline README.md:29]: I like using /api/. Should we make our routes use this naming convention?

File: client/components/statsTrack.jsx [hunk 3]
```diff
@@ -1,18 +1,35 @@
 import React from 'react';
-import Moment from 'moment';
+import moment from 'moment';
 import BackButton from './BackButton';
 import ProgressBar from './ProgressBar';
 import style from '../style.css';
 
-class StatsTrack extends React.Component {
+const HOST = 'http://localhost:3002';
```
Comment (IzzyLeeDev) [inline client/components/statsTrack.jsx:7]: This is quite the heavy refactor of the front-end, but I think you managed to make this file more legible and slightly more modular while maintaining the same functionality. Good job!

File: db/models.js [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
+      .then((result) => result.rows[0])
       .catch((err) => { console.error(err); });
   }
 
-  static getCampaignsByAuthor(db, author) {
-    return db.query('SELECT * FROM campaigns WHERE author = ?', [author])
+  static getCampaignsByUser(db, username) {
+    return db.query('SELECT * FROM campaigns WHERE _user = $1;', [username])
+      .then((result) => result.rows)
       .catch((err) => { console.error(err); });
```
Comment (IzzyLeeDev) [inline db/models.js:12]: `  .catch((err) => console.error(err));`

Definitely a nitpicky suggestion, figured I'd leave it anyway.

File: db/models.js [hunk 5]
```diff
@@ -1,35 +1,36 @@
...
   }
 
   static addCampaign(db, data) {
-    return db.query('INSERT INTO campaigns (campaign, description, author, currency, pledged, goal, backers, endDate, location, _type) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?);', Object.values(data))
+    return db.query('INSERT INTO campaigns (campaign, description, author, _user, country, pledged, goal, backers, endDate, _type) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10);', Object.values(data))
```
Comment (IzzyLeeDev) [inline db/models.js:16]: Why are these hardcoded instead of dynamic? Is this for dev/testing purposes?

**Review by IsaacLeeWebDev — APPROVED** [review]
Overall, I think this is good to go. It looks like you did a big refactor of the front-end code to promote modularity and readability, which I hope helps you save time later on. I want to give you a chance to defend your choice to use react-router-dom, since I'm not too familiar with it. I also would like to further understand your model code, specifically why wildcards have been replaced by explicit dollar amounts. Congratulations on some great progress!

File: README.md (hunk 1 above)
Comment (IamTechknow) [reply README.md:21]: Conventional change - I realized Author refers to a person's name which would include spaces, whereas User refers to a username and would make more sense. This is how kickstarter does it, but my endpoints don't conform to this yet

File: README.md (hunk 2 above)
Comment (IamTechknow) [reply README.md:29]: Yes, I believe using /api/ path to refer to an API is common. I know NASA [does it](https://eonet.sci.gsfc.nasa.gov/docs/v2.1#eventsAPI)!

File: db/models.js (hunk 5 above)
Comment (IamTechknow) [reply db/models.js:16]: Postgres value binding uses numbers, the question marks didn't work for me here unlike MySQL. This is indeed dynamic

**Pull request merged**